php/class-wp-cli.php

```diff
@@ -11,38 +11,225 @@
+	 *
+	 * # Generate guest authors for administrators, editors, authors, and contributors.
+	 * $ wp co-authors-plus create-guest-authors --no-dry-run
+	 * All done! Here are your results:
+	 *
+	 * # Generate guest authors only for authors and freelancers.
+	 * $ wp co-authors-plus create-guest-authors --roles=author,freelancer
+	 * All done! Here are your results:
+	 *
+	 * # Generate guest authors, filtering in batches of 5000.
+	 * $ wp co-authors-plus create-guest-authors --batch-size=5000
+	 * All done! Here are your results:
+	 *
+	 * # This command will fail.
+	 * $ wp co-authors-plus create-guest-authors --roles=author,subscriber
+	 * For performance reasons, subscribers cannot be made guest authors via wp-cli. Please use the admin interface, or pass the argument --force-subscribers.
+	 *
+	 * # This command will succeed. NOTE: Your site may have many subscribers, which may make the command take a long time to run.
+	 * $ wp co-authors-plus create-guest-authors --roles=author,subscriber --force-subscribers
+	 * All done! Here are your results:
+	 *
+	 * # Will actually modify your database and generate guest authors.
+	 * $ wp co-authors-plus create-guest-authors --no-dry-run
+	 * All done! Here are your results:
+	 *
+	 * # Show results of guest author creation attempts as they happen.
+	 * $ wp co-authors-plus create-guest-authors --log-output
+	 * User ID 1 is now linked to Guest Author 2.
+	 * Error while attempting to generate guest author for user ID 3: (error message)
 	 *
 	 * @since 3.0
 	 *
 	 * @subcommand create-guest-authors
+	 * @synopsis [--roles=<roles>] [--batch-size=<batch-size>] [--force-subscribers] [--dry-run] [--log-output]
 	 */
 	public function create_guest_authors( $args, $assoc_args ) {
 		global $coauthors_plus;
 
 		$defaults = array(
-				// There are no arguments at this time
-			);
+			'roles'             => array( 'administrator', 'editor', 'author', 'contributor' ),
+			'batch-size'        => 1000,
+			'force-subscribers' => false,
+			'dry-run'           => true,
+			'log-output'        => false,
+		);
+
 		$this->args = wp_parse_args( $assoc_args, $defaults );
 
-		$users = get_users();
-		$created = 0;
-		$skipped = 0;
-		$progress = \WP_CLI\Utils\make_progress_bar( 'Processing guest authors...', count ( $users ) );
-		foreach ( $users as $user ) {
+		$role_whitelist = $this->args['roles'];
 
-			$result = $coauthors_plus->guest_authors->create_guest_author_from_user_id( $user->ID );
-			if ( is_wp_error( $result ) ) {
-				$skipped++;
-			} else {
-				$created++;
+		if ( ! is_array( $role_whitelist ) ) {
+			$role_whitelist = explode( ',', $role_whitelist );
+		}
+
+		$batch_size = (int) $this->args['batch-size'];
+		if ( $batch_size <= 0 ) {
+			WP_CLI::error( __( 'Batch size must be an integer greater than zero.', 'co-authors-plus' ) );
+		}
+
+		if ( true === $this->args['force-subscribers'] ) {
+			$force_subscribers = true;
+		} else {
+			$force_subscribers = false;
+		}
```

Comment:
At this point we can just have a `$force_subscribers = $this->args['force_subscribers'];` , and the same applies to the other bool options :)